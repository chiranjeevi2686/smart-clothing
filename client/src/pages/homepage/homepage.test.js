import React from 'react';
import { render } from '@testing-library/react';
import HomePage from './homepage.component';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const defaultStore = {
    directory: {
        sections: [{
            id: 1,
            imageUrl: "https://i.ibb.co/cvpntL1/hats.png",
            linkUrl: "shop/hats",
            size: "",
            title: "hats"
        }]
    }
};
const mockStore = configureStore();
describe('HomePage component', () => {
    it('should render HomePage', () => {
        const { asFragment } = render(<Provider store={mockStore(defaultStore)}>
            <HomePage />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
    })
})

