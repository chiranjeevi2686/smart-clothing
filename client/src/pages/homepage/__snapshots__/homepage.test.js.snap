// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should render HomePage 1`] = `
<DocumentFragment>
  <div
    class="sc-gKsewC kniMkA"
  >
    <div
      class="sc-jSgupP gFhuWt"
    >
      <div
        class="sc-bdfBwQ cueKct"
        data-test-id="shop-menu-item"
      >
        <div
          class="sc-gsTCUz CKdSB background-image"
        />
        <div
          class="sc-dlfnbm JasYd"
        >
          <span
            class="sc-hKgILt eLkpOW"
          >
            HATS
          </span>
          <span
            class="sc-eCssSg cAcTCL"
          >
            SHOP NOW
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
